### PR TITLE
added multiple backend exception testing + raised test coverage

### PR DESCRIPTION
**Pull request by armaan-dhillon**
## What
  * Adds tests for multiple backend exception cases
  * Raises test coverage back up to 90%>

## How to Test
  * `pytest tests` within rubicon-ml-dev environment

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -91,7 +91,7 @@ combine_as_imports = True
 markers = 
 	run_notebooks: tests that run Jupyter notebooks
 	write_files: tests that physically write files to local and S3 filesystems
-addopts = --cov=./rubicon_ml --cov-report=term-missing --cov-fail-under=85 -m="not write_files"
```
Comment (ryanSoley) [inline setup.cfg:94]: 🎉

File: tests/unit/client/test_artifact_client.py [hunk 2]
```diff
@@ -26,6 +29,23 @@ def test_get_data(project_client):
     assert artifact.data == data
 
 
+def test_internal_get_data_multiple_backend_error():
```
Comment (ryanSoley) [inline tests/unit/client/test_artifact_client.py:32]: this and `test_get_data_multiple_backend_error` on line 74 seem to be identical - am I missing something?

**Review by ryanSoley — COMMENTED** [review]
looks great! one small thing

File: tests/unit/client/test_artifact_client.py (hunk 2 above)
Comment (armaan-dhillon) [reply tests/unit/client/test_artifact_client.py:32]: Hey! This line `test_get_data_multiple_backend_error` on line 74 is testing `Artifact.get_data()` and the line 84 `test_internal_get_data_multiple_backend_error` is testing the private `Artifact._get_data()`

Comment (ryanSoley) [reply tests/unit/client/test_artifact_client.py:32]: gotcha, thanks for being thorough!

**Review by ryanSoley — APPROVED** [review]
great job! thanks for the additions!

**Pull request merged**